src/Model/Operation/ProductSyncHandler.php

```diff
@@ -231,30 +274,30 @@ private function handleCheapestVariant(
     }
 
     private function handleMainVariant(
-        SalesChannelProductEntity $product,
+        ProductEntity $product,
         VariantListingConfig $variantConfig,
-    ): ProductEntity {
+    ): ?ProductEntity {
         $mainProduct = null;
         $variants = new ProductCollection([$product]);
 
         foreach ($product->getChildren() as $child) {
-            if ($child->getId() === $variantConfig->getMainVariantId()) {
+            if ($child->getId() === $variantConfig->getMainVariantId() && $product->getActive()) {
```

Comment:
Don't we ned to check the child?

```suggestion
            if ($child->getId() === $variantConfig->getMainVariantId() && $child->getActive()) {
```